src/main/java/org/quickfixj/ilink3/ILink3MessageConverter.java

```diff
@@ -302,13 +302,17 @@ public static void convertFromFIXAndSend(Message fixMessage, ILink3Connection co
 	    throw new ILink3ConnectorException("Could not claim buffer for sending message " + fixMessage);
 	}
 
+	Long msqseqnum = connection.nextSentSeqNo() - 1;
 	// fill fields from FIX to SBE message
 	try {
 	    ILink3MessageConverter.convertFromFIX(fixMessage, messageEncoderFlyweight,
-		    connection.nextSentSeqNo() - 1 /* tryClaim already incremented by 1 */);
+				msqseqnum /* tryClaim already incremented by 1 */);
 	} catch (NumberFormatException | InvalidMessage | FieldNotFound e) {
 	    throw new ILink3ConnectorException(e);
 	}
+	//To have a complete quickfix representation of the message we feed it back here. (Currently the seqnum is the important one )
+	fixMessage.setString(SEQ_NUM, String.valueOf(msqseqnum));
```

Comment:
will set this globally in `convertFromFIX()` so no need to return the message

Reply:
for the correct sending.

But what i needed  was a way to flush all the complete messages (including the seqnum) to a db. I'm using the returned message for that, as I'm creating a version without the seqnum in it before i enter this function. It also allows any defaults that are being set by the backend to be returned to the frontend.

(This is also needed for the audittrail for cme, they expect the full human readible fix representation)

Reply:
No problem, will add message as return value also.